File: k9mail/src/main/java/security/pEp/ui/keyimport/KeyImportView.java
package security.pEp.ui.keyimport;

import com.fsck.k9.pEp.PEpProvider;

public interface KeyImportView {

    void showPositiveFeedback();

    void finish();

    void renderDialog(PEpProvider.KeyDetail keyDetail, String from);

    void showNegativeFeedback();
}

File: k9mail/src/main/java/security/pEp/ui/keyimport/KeyImportView.kt
package security.pEp.ui.keyimport

import android.content.Context
import foundation.pEp.jniadapter.Identity

interface KeyImportView {

    fun openFileChooser()

    fun showCorrectKeyImport()

    fun showFailedKeyImport()

    fun finish()

    fun getApplicationContext(): Context

    fun showLoading()

    fun hideLoading()

    fun showKeyImportConfirmationDialog(firstIdentity: Identity, filename: String)

    fun showLayout()
}